### PR TITLE
socketpair: allow EWOULDBLOCK when reading the pair check bytes

### LINKED ISSUE #10561: libcurl 7.88.0- getaddrinfo() thread failed to start
I use libcurl, compiled as a static library, in many of my programs.
Code written in C++, for Windows, using Visual Studio.
libcurl static library is compiled to call DLL Windows SSPI, and DLL WinIDN.
I have used all previous versions from 7.50.3 to 7.86.0, and all works great.

Recently I saw 7.88.0 is available, so I downloaded and compiled that one, using the same settings as all my previous builds.
Now all of my programs fail, with a message:
"getaddrinfo() thread failed to start".

So, I have gone back to using the previous version 7.86.0, and all my programs work again.

What might be causing this bug in 7.88.0?
Is there some extra curl setup required in this new version which I will have to change in my programs to make them work with 7.88.0?

### PR DESCRIPTION
**Pull request by bagder**
Reported-by: Gunamoi Software
Debuggged-by: Jay Satiro
Fixes #10561

### CODE REVIEW

**Comment by bagder** [comment]
@jay this change is kind of made "blindly", it would be super cool if you could verify/adjust it

**Comment by jay** [comment]
fyi I started running repeat.bat in this branch, 20 min no problems so far. I am going to increase MAX and let it run for a while.
~~~
git checkout bagder/socketpair-block
# build curl in visual studio config 'DLL Release - DLL Windows SSPI - DLL WinIDN'
# then checkout just repeat.bat from debug branch and run it from a command prompt
git checkout win32_debug_socketpair_init repeat.bat
~~~

File: lib/socketpair.c [hunk 1]
```diff
@@ -114,13 +114,42 @@ int Curl_socketpair(int domain, int type, int protocol,
...
+#else
+          /* errno may be EWOULDBLOCK or on some systems EAGAIN when it
+             returned due to its inability to send off data without
+             blocking. We therefore treat both error codes the same here */
+          (EWOULDBLOCK == sockerr) || (EAGAIN == sockerr) ||
+          (EINTR == sockerr) || (EINPROGRESS == sockerr)
+#endif
+          ) {
+          /* busy-loop */
+          continue;
```
Comment (jay) [inline lib/socketpair.c:140]: I've been monitoring send results and when it returns wouldblock it often happens hundreds of times in a row, so now i'm testing a slight modification with a poll check beforehand

**Comment by jay** [comment]
I have read through #7144 and #10144, and I don't know if there is a case for checking the source port. It seems to me like Proxifier is the problem. I do see that FF at one point a long time ago checked the source port in PR_NewTCPSocketPair but I don't see a similar check in the [current code](https://searchfox.org/mozilla-central/source/netwerk/base/PollableEvent.cpp#42).

**Comment by bagder** [comment]
did you actually see a case where you need to allow **60** seconds of trying? I would've guessed just a few would be enough...

**Comment by bagder** [comment]
I'll merge this in a sec and go for release

**Comment by jay** [comment]
> did you actually see a case where you need to allow **60** seconds of trying? I would've guessed just a few would be enough...

no. i'm guessing how bad it could be but i didn't see anything like that.

**Pull request closed without merging**